cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -197,6 +201,44 @@ TEST(S3Options, FromUri) {
   ASSERT_RAISES(Invalid, S3Options::FromUri("s3:///foo/bar/", &path));
 }
 
+TEST(S3Options, FromAccessKey) {
+  S3Options options;
+
+  // session token is optional and should default to empty string
+  options = S3Options::FromAccessKey("access", "secret");
+  ASSERT_EQ(options.GetAccessKey(), "access");
+  ASSERT_EQ(options.GetSecretKey(), "secret");
+  ASSERT_EQ(options.GetSessionToken(), "");
+
+  options = S3Options::FromAccessKey("access", "secret", "token");
+  ASSERT_EQ(options.GetAccessKey(), "access");
+  ASSERT_EQ(options.GetSecretKey(), "secret");
+  ASSERT_EQ(options.GetSessionToken(), "token");
+}
+
+TEST(S3Options, FromAssumeRole) {
+  S3Options options;
+
+  // we set this envvar to speed up tests by ensuring
+  // DefaultAWSCredentialsProviderChain does not query (inaccessible)
+  // EC2 metadata endpoint
+  ASSERT_OK(SetEnvVar("AWS_EC2_METADATA_DISABLED", "true"));
```

Comment:
So, the right way to do this would be to create a test fixture:
```c++
class S3OptionsTest : public ::testing::Test {
 public:
  void SetUp() {
    ASSERT_OK(SetEnvVar("AWS_EC2_METADATA_DISABLED", "true"));
  }
  void TearDown() {
    ASSERT_OK(DelEnvVar("AWS_EC2_METADATA_DISABLED"));
  }
};
```
And then to replace every `TEST(S3Options, ...)` with `TEST_F(S3OptionsTest, ...)`.